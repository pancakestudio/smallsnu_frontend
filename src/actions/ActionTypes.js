export const BUILDING_CLICK = "BUILDING_CLICK";
