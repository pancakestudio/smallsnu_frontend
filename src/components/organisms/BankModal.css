.bankModal {
  min-width: 50vw;
}

.bankModal .title {
  margin: auto 0px;
}
