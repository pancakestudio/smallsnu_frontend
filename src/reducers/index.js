import { combineReducers } from 'redux';
import buildingButton from './buildingButton';

const reducers = combineReducers({
  buildingButton
});

export default reducers;
