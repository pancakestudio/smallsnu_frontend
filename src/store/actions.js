import * as types from './actionTypes'

/******** Map ********/

export function requestBuilding(bldgNo){
  return{
    type: types.REQUEST_BUILDING,
    bldgNo
  };
}

export function getBuildingSuccess(krName, bldgNo, info, restaurants, seminars, lectures, posts){
  return{
    type: types.GET_BUILDING_SUCCESS,
    krName,
    bldgNo,
    info,
    restaurants,
    seminars,
    lectures,
    posts,
  }
}

export function getBuildingFailure(error){
  return{
    type: types.GET_BUILDING_FAILURE,
    error
  }
}

export function zoomChanged(zoomLevel){
  return {
    type: types.ZOOM_CHANGED,
    zoomLevel
  }
}

export function hideMarkers(){
  return {
    type: types.HIDE_MARKERS
  }
}

/******** SideBar ********/

export function toggleSideBar(){
  return {
    type: types.TOGGLE_SIDE_BAR
  }
}

/******** SearchBar ********/

export function searchValueChange(bldgNo){
  return{
    type: types.SEARCH_VALUE_CHANGE,
    bldgNo
  };
}

export function search(bldgNo, bldgPos){
  return{
    type: types.SEARCH,
    bldgNo,
    bldgPos
  }
}

/******** Post ********/

export function requestBoard(bldgNo){
  return {
    type: types.REQUEST_BOARD,
    bldgNo
  }
}

export function getBoardSuccess(posts, bldgNo){
  return {
    type: types.GET_BOARD_SUCCESS,
    posts,
    bldgNo
  }
}

export function getBoardFailure(error){
  return {
    type: types.GET_BOARD_FAILURE,
    error
  }
}

export function requestPost(id) {
  return {
    type: types.REQUEST_POST,
    id
  }
}

export function getPostSuccess(post){
  return {
    type: types.GET_POST_SUCCESS,
    post
  }
}

export function getPostFailure(error){
  return {
    type: types.GET_POST_FAILURE,
    error
  }
}

export function deletePost(post, bldgNo){
  return{
    type: types.DELETE_POST,
    post,
    bldgNo,
  }
}

export function savePost(post, bldgNo){
  return{
    type: types.SAVE_POST,
    post,
    bldgNo
  }
}

export function editPost(post, bldgNo){
  return{
    type: types.EDIT_POST,
    post,
    bldgNo
  }
}

export function editPostFlag(){
  return{
    type: types.EDIT_POST_FLAG,
  }
}

export function showWritePost() {
  return{
    type: types.SHOW_WRITE_POST,
  };
}

export function showPostPWCheck(){
  return{
    type: types.SHOW_POST_PW_CHECK
  }
}

export function hidePostPWCheck(){
  return{
    type: types.HIDE_POST_PW_CHECK
  }
}

export function postPWCheck(password) {
  return{
    type: types.POST_PW_CHECK,
    password
  }
}

export function changeBoardPage(page){
  return {
    type: types.CHANGE_BOARD_PAGE,
    page
  }
}

/******** Restaurant ********/

export function toggleResMarker(){
  return{
    type: types.TOGGLE_RES_MARKER,
  }
}

export function requestRestaurant(id){
  return{
    type: types.REQUEST_RESTAURANT,
    id
  }
}

export function getRestaurantSuccess(restaurant){
  return{
    type : types.GET_RESTAURANT_SUCCESS,
    restaurant,
  }
}

export function getRestaurantFailure(error){
  return{
    type: types.GET_RESTAURANT_FAILURE,
    error
  }
}

export function requestAllRestaurants(){
  return{
    type: types.REQUEST_ALL_RESTAURANTS,
  }
}


export function getAllRestaurantsSuccess(restaurants){
  return{
    type : types.GET_ALL_RESTAURANTS_SUCCESS,
    restaurants,
  }
}

export function getAllRestaurantsFailure(error){
  return{
    type: types.GET_ALL_RESTAURANTS_FAILURE,
    error
  }
}

/******** Seminar ********/

export function toggleSemiMarker(){
  return{
    type: types.TOGGLE_SEMI_MARKER,
  }
}

export function changeSeminarPage(page){
  return {
    type: types.CHANGE_SEMINAR_PAGE,
    page
  }
}

export function requestSeminar(id){
  return {
    type: types.REQUEST_SEMINAR,
    id
  }
}

export function getSeminarSuccess(seminar){
  return {
    type: types.GET_SEMINAR_SUCCESS,
    seminar
  }
}

export function getSeminarFailure(error){
  return {
    type: types.GET_SEMINAR_FAILURE,
    error
  }
}

export function requestBldgSeminars(bldgNo){
  return {
    type: types.REQUEST_BLDG_SEMINARS,
    bldgNo
  }
}

export function getBldgSeminarsSuccess(seminars, bldgNo){
  return {
    type: types.GET_BLDG_SEMINARS_SUCCESS,
    seminars,
    bldgNo
  }
}

export function getBldgSeminarsFailure(error){
  return {
    type: types.GET_BLDG_SEMINARS_FAILURE,
    error
  }
}

export function requestAllSeminars(){
  return {
    type: types.REQUEST_ALL_SEMINARS,
  }
}

export function getAllSeminarsSuccess(seminars){
  return {
    type: types.GET_ALL_SEMINARS_SUCCESS,
    seminars
  }
}

export function getAllSeminarsFailure(error){
  return {
    type: types.GET_ALL_SEMINARS_FAILURE,
    error
  }
}

/******** Bank ********/

export function toggleBankMarker(){
  return{
    type: types.TOGGLE_BANK_MARKER,
  }
}

export function requestAllBanks(){
  return{
    type: types.REQUEST_ALL_BANKS,
  }
}

export function getAllBanksSuccess(banks){
  return{
    type: types.GET_ALL_BANKS_SUCCESS,
    banks,
  }
}

export function getAllBanksFailure(error){
  return{
    type: types.GET_ALL_BANKS_FAILURE,
    error,
  }
}

export function requestBank(id){
  return{
    type: types.REQUEST_BANK,
    id
  }
}

export function getBankSuccess(bank){
  return{
    type: types.GET_BANK_SUCCESS,
    bank
  }
}

export function getBankFailure(error){
  return{
    type: types.GET_BANK_FAILURE,
    error
  }
}


/******** ATM ********/

export function toggleATMMarker(){
  return{
    type: types.TOGGLE_ATM_MARKER,
  }
}

export function requestAllATMs(){
  return{
    type: types.REQUEST_ALL_ATMS,
  }
}

export function getAllATMsSuccess(atms){
  return{
    type: types.GET_ALL_ATMS_SUCCESS,
    atms,
  }
}

export function getAllATMsFailure(error){
  return{
    type: types.GET_ALL_ATMS_FAILURE,
    error,
  }
}

export function requestATM(id){
  return{
    type: types.REQUEST_ATM,
    id
  }
}

export function getATMSuccess(atm){
  return{
    type: types.GET_ATM_SUCCESS,
    atm
  }
}

export function getATMFailure(error){
  return{
    type: types.GET_ATM_FAILURE,
    error
  }
}

/******** Cafe ********/

export function toggleCafeMarker(){
  return{
    type: types.TOGGLE_CAFE_MARKER,
  }
}

export function requestAllCafes(){
  return{
    type: types.REQUEST_ALL_CAFES,
  }
}

export function getAllCafesSuccess(cafes){
  return{
    type: types.GET_ALL_CAFES_SUCCESS,
    cafes,
  }
}

export function getAllCafesFailure(error){
  return{
    type: types.GET_ALL_CAFES_FAILURE,
    error,
  }
}

export function requestCafe(id){
  return{
    type: types.REQUEST_CAFE,
    id
  }
}

export function getCafeSuccess(cafe){
  return{
    type: types.GET_CAFE_SUCCESS,
    cafe
  }
}

export function getCafeFailure(error){
  return{
    type: types.GET_CAFE_FAILURE,
    error
  }
}


/******** Conv ********/

export function toggleConvMarker(){
  return{
    type: types.TOGGLE_CONV_MARKER,
  }
}

export function requestAllConves(){
  return{
    type: types.REQUEST_ALL_CONVES,
  }
}

export function getAllConvesSuccess(conves){
  return{
    type: types.GET_ALL_CONVES_SUCCESS,
    conves,
  }
}

export function getAllConvesFailure(error){
  return{
    type: types.GET_ALL_CONVES_FAILURE,
    error,
  }
}

export function requestConv(id){
  return{
    type: types.REQUEST_CONV,
    id
  }
}

export function getConvSuccess(conv){
  return{
    type: types.GET_CONV_SUCCESS,
    conv
  }
}

export function getConvFailure(error){
  return{
    type: types.GET_CONV_FAILURE,
    error
  }
}
