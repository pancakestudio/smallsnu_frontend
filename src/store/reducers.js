import { combineReducers } from 'redux';
import * as types from './actionTypes'

/******** Map ********/

function currentPos(state = {lat: 37.459, lng: 126.952}, action){
  switch(action.type){
    case types.SEARCH:
      return action.bldgPos
    default:
      return state
  }
}

function zoom(state = 17, action) {
  switch(action.type){
    case types.TOGGLE_RES_MARKER:
    case types.TOGGLE_SEMI_MARKER:
      return 15
    case types.ZOOM_CHANGED:
      return action.zoomLevel
    default:
      return state
  }
}

function selectedBldg(state = {}, action){
  switch(action.type){
    case types.GET_BUILDING_SUCCESS:
      return Object.assign({}, state, {
        krName: action.krName,
        bldgNo: action.bldgNo,
        info: action.info,
        restaurants: action.restaurants,
        seminars: action.seminars,
        posts: action.posts
      })
    case types.GET_BOARD_SUCCESS:
      return Object.assign({}, state, {
        posts: action.posts
      })
    default:
      return state
  }
}

function showSearchMarker(state = false, action){
  switch(action.type){
    case types.SEARCH:
      return true
    default:
      return false
  }
}

function showResMarkers(state = false, action){
  switch(action.type){
    case types.TOGGLE_RES_MARKER:
      return !state
    case types.GET_ALL_RESTAURANTS_FAILURE:
    case types.HIDE_MARKERS:
    case types.TOGGLE_SEMI_MARKER:
    case types.TOGGLE_BANK_MARKER:
    case types.TOGGLE_ATM_MARKER:
    case types.TOGGLE_CAFE_MARKER:
    case types.TOGGLE_CONV_MARKER:
      return false
    default:
      return state
  }
}

function showBankMarkers(state = false, action){
  switch(action.type){
    case types.TOGGLE_BANK_MARKER:
      return !state
    case types.GET_ALL_BANKS_FAILURE:
    case types.HIDE_MARKERS:
    case types.TOGGLE_SEMI_MARKER:
    case types.TOGGLE_RES_MARKER:
    case types.TOGGLE_ATM_MARKER:
    case types.TOGGLE_CAFE_MARKER:
    case types.TOGGLE_CONV_MARKER:
      return false
    default:
      return state
  }
}

function showATMMarkers(state = false, action){
  switch(action.type){
    case types.TOGGLE_ATM_MARKER:
      return !state
    case types.GET_ALL_ATMS_FAILURE:
    case types.HIDE_MARKERS:
    case types.TOGGLE_SEMI_MARKER:
    case types.TOGGLE_RES_MARKER:
    case types.TOGGLE_BANK_MARKER:
    case types.TOGGLE_CAFE_MARKER:
    case types.TOGGLE_CONV_MARKER:
      return false
    default:
      return state
  }
}

function showCafeMarkers(state = false, action){
  switch(action.type){
    case types.TOGGLE_CAFE_MARKER:
      return !state
    case types.GET_ALL_CAFES_FAILURE:
    case types.HIDE_MARKERS:
    case types.TOGGLE_SEMI_MARKER:
    case types.TOGGLE_RES_MARKER:
    case types.TOGGLE_BANK_MARKER:
    case types.TOGGLE_ATM_MARKER:
    case types.TOGGLE_CONV_MARKER:
      return false
    default:
      return state
  }
}

function showConvMarkers(state = false, action){
  switch(action.type){
    case types.TOGGLE_CONV_MARKER:
      return !state
    case types.GET_ALL_CONVES_FAILURE:
    case types.HIDE_MARKERS:
    case types.TOGGLE_SEMI_MARKER:
    case types.TOGGLE_RES_MARKER:
    case types.TOGGLE_BANK_MARKER:
    case types.TOGGLE_ATM_MARKER:
    case types.TOGGLE_CAFE_MARKER:
      return false
    default:
      return state
  }
}

function showSemiMarkers(state = false, action){
  switch(action.type){
    case types.TOGGLE_SEMI_MARKER:
      return !state
    case types.GET_ALL_SEMINARS_FAILURE:
    case types.HIDE_MARKERS:
    case types.TOGGLE_RES_MARKER:
      return false
    default:
      return state
  }
}

function searchedBldg(state = "0", action){
  switch(action.type){
    case types.SEARCH:
      return action.bldgNo
    default:
      return state
  }
}

/******** SideBar ********/

function showSideBar(state = false, action){
  switch(action.type){
    case types.TOGGLE_SIDE_BAR:
      return !state
    default:
      return state
  }
}

/******** SearchBar ********/

function searchingBldg(state = "0", action){
  switch(action.type){
    case types.SEARCH_VALUE_CHANGE:
      return action.bldgNo
    default:
      return state
  }
}


/******** Post ********/

function showWritePostModal(state = false, action){
  switch(action.type){
    case types.SHOW_WRITE_POST:
      return true
    default:
      return false
  }
}

function selectedBoardBldgNo(state = "0", action){
  switch(action.type){
    case types.GET_BOARD_SUCCESS:
    case types.GET_BUILDING_SUCCESS:
      return action.bldgNo
    default:
      return state
  }
}

function selectedPostList(state = [], action){
  switch(action.type){
    case types.GET_BOARD_SUCCESS:
      return action.posts
    default:
      return state
  }
}

function selectedPost(state = {}, action){
  switch(action.type){
    case types.GET_POST_SUCCESS:
      return action.post
    default:
      return state
  }
}

function isEdit(state = false, action){
  switch(action.type){
    case types.EDIT_POST_FLAG:
      return true
    case types.GET_BOARD_SUCCESS:
      return false
    default:
      return state
  }
}

function postPW(state = "", action){
  switch(action.type){
    case types.POST_PW_CHECK:
      return action.password
    case types.GET_BOARD_SUCCESS:
      return ""
    default:
      return state
  }
}

function showPostPWCheck(state = false, action){
  switch(action.type){
    case types.SHOW_POST_PW_CHECK:
      return true
    default:
      return false
  }
}

function activeBoardPage(state = 1, action) {
  switch(action.type){
    case types.CHANGE_BOARD_PAGE:
      return action.page
    case types.GET_BOARD_SUCCESS:
      return 1
    default:
      return state
  }
}
/******** Bank ********/

function allBanks(state = [], action){
  switch(action.type){
    case types.GET_ALL_BANKS_SUCCESS:
      return action.banks
    default:
      return state
  }
}

function selectedBank(state = {}, action){
  switch(action.type){
    case types.GET_BANK_SUCCESS:
      return action.bank
    default:
      return state
  }
}

/******** ATM ********/

function allATMs(state = [], action){
  switch(action.type){
    case types.GET_ALL_ATMS_SUCCESS:
      return action.atms
    default:
      return state
  }
}

function selectedATM(state = {}, action){
  switch(action.type){
    case types.GET_ATM_SUCCESS:
      return action.atm
    default:
      return state
  }
}

/******** Cafe ********/

function allCafes(state = [], action){
  switch(action.type){
    case types.GET_ALL_CAFES_SUCCESS:
      console.log(action.cafes)
      return action.cafes
    default:
      return state
  }
}

function selectedCafe(state = {}, action){
  switch(action.type){
    case types.GET_CAFE_SUCCESS:
      return action.cafe
    default:
      return state
  }
}

/******** Convenient Store ********/

function allConves(state = [], action){
  switch(action.type){
    case types.GET_ALL_CONVES_SUCCESS:
      return action.conves
    default:
      return state
  }
}

function selectedConv(state = {}, action){
  switch(action.type){
    case types.GET_CONV_SUCCESS:
      return action.conv
    default:
      return state
  }
}

/******** Restaurant ********/

function selectedRes(state = {}, action){
  switch(action.type){
    case types.GET_RESTAURANT_SUCCESS:
      return action.restaurant
    default:
      return state
  }
}

function allRestaurants(state = [], action){
  switch(action.type){
    case types.GET_ALL_RESTAURANTS_SUCCESS:
      return action.restaurants
    default:
      return state
  }
}

/******** Seminar ********/

function selectedSemi(state = {}, action){
  switch(action.type){
    case types.GET_SEMINAR_SUCCESS:
      return action.seminar
    default:
      return state
  }
}

function selectedSemiList(state = [], action){
  switch(action.type){
    case types.GET_BLDG_SEMINARS_SUCCESS:
      return action.seminars
    default:
      return state
  }
}

function selectedSemiListBldgNo(state = "0", action){
  switch(action.type){
    case types.GET_BLDG_SEMINARS_SUCCESS:
      return action.bldgNo
    default:
      return state
  }
}

function activeSemiPage(state = 1, action) {
  switch(action.type){
    case types.CHANGE_SEMINAR_PAGE:
      return action.page
    case types.GET_BLDG_SEMINARS_SUCCESS:
      return 1
    default:
      return state
  }
}


function allSeminars(state = [], action) {
  switch(action.type){
    case types.GET_ALL_SEMINARS_SUCCESS:
      return action.seminars
    default:
      return state
  }
}

/******** App ********/

function error(state = "", action){
  switch(action.type){
    case types.GET_BUILDING_FAILURE:
    case types.GET_BOARD_FAILURE:
    case types.GET_POST_FAILURE:
    case types.GET_RESTAURANT_FAILURE:
    case types.GET_ALL_RESTAURANTS_FAILURE:
    case types.GET_SEMINAR_FAILURE:
    case types.GET_BLDG_SEMINARS_FAILURE:
    case types.GET_ALL_SEMINARS_FAILURE:
      return action.error
    default:
      return ""
  }
}

const reducers = combineReducers({

  currentPos, zoom, selectedBldg, showSearchMarker, showResMarkers,
  showSemiMarkers, searchedBldg, showBankMarkers, showATMMarkers, showCafeMarkers,
  showConvMarkers,// Map

  showSideBar, // SideBar
  searchingBldg, // SearchBar

  showWritePostModal, selectedBoardBldgNo, selectedPostList, selectedPost,
  isEdit, postPW, showPostPWCheck, activeBoardPage, // Post

  allBanks,selectedBank, // Bank
  allATMs, selectedATM, // ATM
  allCafes, selectedCafe, // Cafe
  allConves, selectedConv,  // Convenient Store
  selectedRes, allRestaurants, // Restaurant

  selectedSemi, selectedSemiList, selectedSemiListBldgNo,
  activeSemiPage, allSeminars, // Seminar

  error // App

});

export default reducers
