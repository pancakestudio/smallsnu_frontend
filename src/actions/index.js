import * as types from './ActionTypes';

export function buildingClick(e){
  return{
    type: types.BUILDING_CLICK,
    e
  };
}
