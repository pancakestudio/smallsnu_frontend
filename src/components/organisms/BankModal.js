import React from 'react'
import { Card, ListGroup, Modal, Button } from 'react-bootstrap'
import { FaAngleLeft } from 'react-icons/fa'
import { historyPush } from '../../utils/Functions'
import './BankModal.css'

export const BankModal = ({bank}) => {
  const handleBack = () => {
    if(bank && bank.building){
      let bldgNo = bank.building.code
      historyPush(`/building/${bldgNo}`)
    } else {
      historyPush('/')
    }
  }
  let header, body
  if(bank && bank.building){
    header = (
      <Modal.Header closeButton>
        <Button className="back" onClick={handleBack}><FaAngleLeft /></Button>
        <Modal.Title className="title">{bank.kr_name}</Modal.Title>
      </Modal.Header>
    )
    body = (
      <Modal.Body className="modalBody">
        <Card className="op_hours border-0">
          <ListGroup variant="flush">
            <ListGroup.Item className="border-left-0 border-right-0 border-top-0"><strong>위치:</strong> {bank.building.kr_name}</ListGroup.Item>
            <ListGroup.Item className="border-left-0 border-right-0 border-bottom-0"><strong>운영 시간:</strong> {bank.operating_hours}</ListGroup.Item>
          </ListGroup>
        </Card>
      </Modal.Body>
    )
  } else {
    header = (
      <Modal.Header closeButton>
        <Button className="back" onClick={handleBack}><FaAngleLeft /></Button>
        <Modal.Title>은행 정보가 없습니다.</Modal.Title>
      </Modal.Header>
    )
    body = (
      <Modal.Body>
        정보없음.
      </Modal.Body>
    )
  }
 return(
   <Modal
     show = {true}
     onHide = {()=>{historyPush('/')}}
     dialogClassName = "bankModal"
     scrollable = {true}
     centered
   >
     {header}
     {body}
   </Modal>
 )
}
