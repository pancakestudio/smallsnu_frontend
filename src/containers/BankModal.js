import { connect } from 'react-redux'
import { BankModal } from '../components/organisms/BankModal'

const mapStateToProps = (state) => {
  return{
    bank : state.selectedBank,
  }
}

export default connect(mapStateToProps, null)(BankModal)
