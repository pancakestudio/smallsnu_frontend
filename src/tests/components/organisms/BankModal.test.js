import React from 'react'
import { Provider } from 'react-redux'
import { BankModal } from '../../../components/organisms/BankModal'
import ConnectedBankModal from '../../../containers/BankModal'
import configureStore from 'redux-mock-store'
import { shallow, mount } from 'enzyme'
import reducers from '../../../store/reducers'
import { createStore } from 'redux'
import * as actions from '../../../store/actions'

describe('BankModal', () => {
  let component

  it('renders correctly', () => {
    component = shallow(
      <BankModal
      bank={{kr_name: '은행', bldgNo:'60',
        building:{
          code: "60",
          kr_name: "행정관(60동)"}}}
      show = {true}
      />
    )
  })

  it('matches snapshot', ()=>{
    expect(component).toMatchSnapshot()
  })

  it('has a modal', ()=>{
    expect(component.find('Bootstrap(Modal)').exists()).toBe(true)
  })

  it('has a correct Bank name', ()=>{
    expect(component.find('ModalTitle').text()).toBe('은행')
  })

  it('has a correct building name', ()=>{
    expect(component.find('Bootstrap(ListGroupItem)').at(0).text()).toBe('위치: 행정관(60동)')
  })

  it('data is given null', () => {
    component.setProps({bank: null})
    expect(component.find('ModalTitle').text()).toBe('은행 정보가 없습니다.')
  })

  it('calls functions', ()=>{
    component.find('Bootstrap(Modal)').simulate('hide')
    expect(global.window.location.pathname).toEqual('/')
  })

})

describe('ConnectedBankModal', () => {
  const initialState = {
    selectedBank: {
      kr_name: '은행',
      bldgNo: '60',
      building:{
        code: "60",
        kr_name: "행정관(60동)"}
    }
  }
  const mockStore = configureStore()
  let store, component

  it('renders correctly', ()=>{
    store = mockStore(initialState)
    component = mount(<Provider store={store}><ConnectedBankModal/></Provider>)
  })

  it('matches snapshot', ()=>{
    expect(component).toMatchSnapshot()
  })

  it('has a modal', ()=>{
    expect(component.find('Bootstrap(Modal)').exists()).toBe(true)
  })

  it('has a correct Bank name', ()=>{
    expect(component.find('ModalTitle').text()).toBe('은행')
  })

})
