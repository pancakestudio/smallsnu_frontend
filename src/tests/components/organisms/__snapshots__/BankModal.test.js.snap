// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BankModal matches snapshot 1`] = `ShallowWrapper {}`;

exports[`ConnectedBankModal matches snapshot 1`] = `ReactWrapper {}`;
